Source/VisualStudioTools/Private/VSTestAdapterCommandlet.cpp

```diff
@@ -2,134 +2,153 @@
+static int32 RunTests(const FString& TestListFile, const FString& ResultsFile)
 {
-	std::wofstream OutFile( *ResultsFile );
-	if ( !OutFile.good() )
+	std::wofstream OutFile(*ResultsFile);
+	if (!OutFile.good())
 	{
-		UE_LOG( LogVisualStudioTools, Error, TEXT( "Failed to open file at path: %s" ), *ResultsFile );
+		UE_LOG(LogVisualStudioTools, Error, TEXT("Failed to open file at path: %s"), *ResultsFile);
 		return 1;
 	}
 
 	TArray< FAutomationTestInfo > TestInfos;
-	if ( TestListFile.Equals( TEXT( "All" ), ESearchCase::IgnoreCase ) )
+	if (TestListFile.Equals(TEXT("All"), ESearchCase::IgnoreCase))
 	{
-		GetAllTests( TestInfos );
+		GetAllTests(TestInfos);
 	}
 	else
 	{
-		ReadTestsFromFile( TestListFile, TestInfos );
+		ReadTestsFromFile(TestListFile, TestInfos);
 	}
 
 	bool AllSuccessful = true;
 
 	FAutomationTestFramework& Framework = FAutomationTestFramework::GetInstance();
 
-	for ( const FAutomationTestInfo& TestInfo : TestInfos )
+	for (const FAutomationTestInfo& TestInfo : TestInfos)
 	{
 		const FString TestCommand = TestInfo.GetTestName();
 		const FString DisplayName = TestInfo.GetDisplayName();
 
-		UE_LOG( LogVisualStudioTools, Log, TEXT( "Running %s" ), *DisplayName );
+		UE_LOG(LogVisualStudioTools, Log, TEXT("Running %s"), *DisplayName);
 
 		const int32 RoleIndex = 0; // always default to "local" role index.  Only used for multi-participant tests
-		Framework.StartTestByName( TestCommand, RoleIndex );
+		Framework.StartTestByName(TestCommand, RoleIndex);
 
 		FDateTime Last = FDateTime::UtcNow();
 
-		while ( !Framework.ExecuteLatentCommands() )
+		while (!Framework.ExecuteLatentCommands())
 		{
 			// Because we are not 'ticked' by the Engine we need to pump the TaskGraph
-			FTaskGraphInterface::Get().ProcessThreadUntilIdle( ENamedThreads::GameThread );
+			FTaskGraphInterface::Get().ProcessThreadUntilIdle(ENamedThreads::GameThread);
 
 			const FDateTime Now = FDateTime::UtcNow();
 			const FTimespan Delta = Now - Last;
 
 			// .. and the core FTicker
-			FTSTicker::GetCoreTicker().Tick( Delta.GetTotalSeconds() );
+			FTSTicker::GetCoreTicker().Tick(Delta.GetTotalSeconds());
 
 			Last = Now;
 		}
 
 		FAutomationTestExecutionInfo ExecutionInfo;
-		const bool CurrentTestSuccessful = Framework.StopTest( ExecutionInfo );
+		const bool CurrentTestSuccessful = Framework.StopTest(ExecutionInfo) && ExecutionInfo.GetErrorTotal() == 0;
 		AllSuccessful = AllSuccessful && CurrentTestSuccessful;
 
-		const FString Result = CurrentTestSuccessful ? TEXT( "OK" ) : TEXT( "FAIL" );
+		const FString Result = CurrentTestSuccessful ? TEXT("OK") : TEXT("FAIL");
 
-		OutFile << TEXT( "[RUNTEST]" ) << *TestCommand << TEXT( "," ) << *DisplayName << TEXT( "," ) << *Result << TEXT( "," ) << ExecutionInfo.Duration << std::endl;
+		OutFile << TEXT("[RUNTEST]") << *TestCommand << TEXT("|") << *DisplayName << TEXT("|") << *Result << TEXT("|") << ExecutionInfo.Duration << std::endl;
```

Comment:
Minor: if this `"[RunTest]"` part is required by VS to parse the line correctly, consider adding a comment so anyone with the source would know it's part of the protocol. Changing serialized data is always prone to break something, but it's good to make it explicit.

Reply:
Will address in future PR soon.